libraries/RC_Channel/RC_Channels.cpp

```diff
@@ -201,6 +201,25 @@ void RC_Channels::read_mode_switch()
     c->read_mode_switch();
 }
 
+/*
+  get the RC input PWM value given a channel number.  Note that
+  channel numbers start at 1, as this API is designed for use in
+  LUA
+*/
+bool RC_Channels::get_pwm(uint8_t c, uint16_t &pwm) const
+{
+    if (c < 1 || c > NUM_RC_CHANNELS) {
+        return false;
+    }
+    RC_Channel *chan = rc_channel(c-1);
```

Comment:
We should be doing a nullptr check on the `chan`, rather then assuming we got one that was valid because we used the defines above, since it's up to the vehicle. I'd remove the entire range check above and just pass c - 1 into the call and check for the nullptr, maybe retraining `c < 1`, but it's not worth the rest of it. Particularly since the actual code path for this is from scripting, which means the `c` will be checked 3 times for being a valid index (scripting, here, and in the `RC_Channel::channel()` call).

Reply:
fixed, and also fixed this related rather nasty bug!
https://github.com/ArduPilot/ardupilot/pull/13081

Reply:
Yikes! Good catch.